src/codegen/c.zig

```diff
@@ -19,24 +21,31 @@ fn map(allocator: *std.mem.Allocator, name: []const u8) ![]const u8 {
     return allocator.dupe(u8, name);
 }
 
-fn renderType(ctx: *Context, writer: std.ArrayList(u8).Writer, T: Type) !void {
+fn renderType(
+    allocator: *std.mem.Allocator,
+    error_msg: **ErrorMsg,
+    header_info: *HeaderInfo,
+    writer: std.ArrayList(u8).Writer,
+    decl: *Decl,
+    T: Type,
+) !void {
     switch (T.zigTypeTag()) {
         .NoReturn => {
             try writer.writeAll("zig_noreturn void");
         },
         .Void => try writer.writeAll("void"),
         .Int => {
             if (T.tag() == .u8) {
-                ctx.file.need_stdint = true;
+                header_info.need_stdint = true;
                 try writer.writeAll("uint8_t");
             } else if (T.tag() == .usize) {
-                ctx.file.need_stddef = true;
+                header_info.need_stddef = true;
                 try writer.writeAll("size_t");
             } else {
-                return ctx.file.fail(ctx.decl.src(), "TODO implement int types", .{});
+                return C.failImpl(allocator, error_msg, decl.src(), "TODO implement int types", .{});
```

Comment:
Not sure whether this is the right way to do this. I wanted to make this independent of `Context` but unfortunately, it's caused a bunch of ugly parameters to get piped through these helpers.

Reply:
If you need to, you could add a `ctx.fail` function which handles it, but we need to decide how errors should be handled here before worrying about the implementation anyways.

Reply:
My bad, I misunderstood where `Context` is coming from. I'll revert this portion of the change.